pkg/api/types.go

```diff
@@ -139,6 +142,21 @@ type PipelineImageCacheStepConfiguration struct {
 	Commands string `json:"commands"`
 }
 
+type TestStepConfiguration struct {
+	// As is the name of the test.
+	As string `json:"as"`
+	// From is the image stream tag in the pipeline to run this
+	// command in.
+	From PipelineImageStreamTagReference `json:"from"`
+	// Commands are the shell commands to run in
+	// the repository root to execute tests.
+	Commands string `json:"commands"`
```

Comment:
Why is it on us to do word splitting?

Reply:
Isn't it? We take this single string and need to feed it into a `[]string{}` for the Pod, so isn't every entry in that array treated as a single arg?

Reply:
We’re feeding a scriptlet to bash as arg[2].  The same as the build command.